app/controllers/middleware_server_controller.rb

```diff
@@ -50,17 +50,24 @@ class MiddlewareServerController < ApplicationController
 
   def add_deployment
     selected_servers = identify_selected_entities
+    deployment_name = params["runtimeName"]
 
-    params[:file] = {
-      :file         => params["file"],
-      :enabled      => params["enabled"],
-      :runtime_name => params["runtimeName"]
-    }
-    run_server_operation(OPERATIONS.fetch(:middleware_add_deployment), selected_servers)
+    existing_deployment = MiddlewareDeployment.find_by(:name => deployment_name, :server_id => selected_servers)
 
-    render :update do |page|
-      page << javascript_prologue
-      page.replace("flash_msg_div", :partial => "layouts/flash_msg")
+    if existing_deployment
+      render :json => {
+        :status => :warn, :msg => _("Deployment \"%s\" already exists on this server.") % deployment_name
```

Comment:
I would fix it for clarity, I think it's a tiny fix so you can do it here or in another PR - up to you